### PR TITLE
Added stackoverflow python job crawler, Get all of job links that con…

### PR DESCRIPTION
**Pull request by sonechca**
…sist of JSON format

### **Describe your change:**

* [x ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by mrmaxguns** [comment]
Here is why the pre-commit failed:

1. Fix end of files: Your file must end with one empty line
2. Trim trailing whitespace: Make sure you have no tabs or spaces at the end of lines. Blank lines should have no tabs or spaces.
3. Black: make sure to run the `black` linter. Black will automatically fix most of the issues described in this list.
4. Flake8: there were two errors:
   1. `web_programming/crawl_stackoverflow_job.py:5:7: F541 f-string is missing placeholders`: your f-string on line 5 is missing placeholders. Replace it with a regular string.
   2. `web_programming/crawl_stackoverflow_job.py:8:1: E302 expected 2 blank lines, found 1`: you need two blank lines before the function.

File: web_programming/crawl_stackoverflow_job.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import requests #Navigating with Python
+from bs4 import BeautifulSoup #extract Stackoverflow pages
+
+#Get the Stackoverflow URL
+URL = f"https://stackoverflow.com/jobs?q=python&sort=i"
```
Comment (mrmaxguns) [inline web_programming/crawl_stackoverflow_job.py:5]: Change the f-string to a regular string

File: web_programming/crawl_stackoverflow_job.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#Collect the stackoverflow information and return main.py
+def get_jobs():
+    last_page = get_last_page()
+    jobs = extract_jobs(last_page)
+    print(jobs)
+
+if __name__ == "__main__":
+    print("Python job finding now...")
+    get_jobs()
```
Comment (mrmaxguns) [inline web_programming/crawl_stackoverflow_job.py:35]: Add a blank line at the end of the file

File: web_programming/crawl_stackoverflow_job.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import requests #Navigating with Python
+from bs4 import BeautifulSoup #extract Stackoverflow pages
+
+#Get the Stackoverflow URL
+URL = f"https://stackoverflow.com/jobs?q=python&sort=i"
+
```
Comment (mrmaxguns) [inline web_programming/crawl_stackoverflow_job.py:6]: Add another empty line before `get_last_page`

File: web_programming/crawl_stackoverflow_job.py (hunk 2 above)
Comment (cclauss) [reply web_programming/crawl_stackoverflow_job.py:35]: Will psf/black add this automatically?

File: web_programming/crawl_stackoverflow_job.py (hunk 3 above)
Comment (cclauss) [reply web_programming/crawl_stackoverflow_job.py:6]: Will psf/black add this automatically?

File: web_programming/crawl_stackoverflow_job.py (hunk 2 above)
Comment (mrmaxguns) [reply web_programming/crawl_stackoverflow_job.py:35]: I believe so

File: web_programming/crawl_stackoverflow_job.py (hunk 3 above)
Comment (mrmaxguns) [reply web_programming/crawl_stackoverflow_job.py:6]: I believe it will

**Comment by xcodz-dot** [comment]
simple fix:
* `pip install pre-commit`
* `pre-commit run -a` in root of local cloned repo

**Pull request closed without merging**